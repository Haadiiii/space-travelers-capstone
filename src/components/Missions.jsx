import React, { useState } from 'react';
import "./missions.css";

export default function Missions(props) {
  const [active, setActive] = useState(false);

  const handleClick = () => {
    setActive((current) => !current);
  };
  return (
    <div className="mission-list">
      <div className="list">
        <div className="list-h"><h1>{props.name}</h1></div>
        <hr />
        <div className="list-p"><p>{props.discription}</p></div>
        <hr />
        <div className="list-s">
          <h3 style={{
            backgroundColor: active ? "rgb(19, 138, 223)" : "",
          }}
          >
            {props.status}
          </h3>
        </div>
        <hr />
        <button className="btn" onClick={handleClick} type="button">Join mission</button>
      </div>
      <hr />
    </div>
  );
}
