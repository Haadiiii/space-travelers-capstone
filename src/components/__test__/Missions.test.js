import React from "react";
import renderer from "react-test-renderer";
import Missions from "./__mock__/missions";

describe('A tests for my missions component', () => {
  it('Check whether mission component renders', () => {
    const mission = {
      id: 1,
      name: 'Thaicom',
      description: 'Thaicom is the name of a series of  communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
      reserved: false,
    };
    const missionTree = renderer.create(
      <Missions
        key={mission.id}
        description={mission.description}
        name={mission.name}
        reserved={mission.reserved}
      />,
    ).toJSON();

    expect(missionTree).toMatchSnapshot();
  });
});
