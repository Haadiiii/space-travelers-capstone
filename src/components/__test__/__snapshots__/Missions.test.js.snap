// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A tests for my missions component Check whether mission component renders 1`] = `
<div
  className="mission-list"
>
  <div
    className="list"
  >
    <div
      className="list-h"
    >
      <h1>
        Thaicom
      </h1>
    </div>
    <hr />
    <div
      className="list-p"
    >
      <p />
    </div>
    <hr />
    <div
      className="list-s"
    >
      <h3
        style={
          {
            "backgroundColor": "rgb(19, 138, 223)",
          }
        }
      >
        Active member
      </h3>
    </div>
    <hr />
    <button
      className="btn"
      style={
        {
          "border": "1px solid red",
          "color": "red",
        }
      }
      type="button"
    >
      Leave Mission
    </button>
  </div>
  <hr />
</div>
`;
